src/System.IO.Ports/tests/SerialPort/DosDevices.cs

```diff
@@ -139,7 +139,7 @@ private static char[] CallQueryDosDevice(string name, out int dataSize)
         public const int ERROR_INSUFFICIENT_BUFFER = 122;
         public const int ERROR_MORE_DATA = 234;
 
-        [DllImport("Kernel32.dll", SetLastError = true, CharSet = CharSet.Auto)]
+        [DllImport("Kernel32.dll", SetLastError = true, EntryPoint = "QueryDosDeviceW", CharSet = CharSet.Unicode)]
```

Comment:
Why is EntryPoint required? We often don't add it, I think. The interop infra tries the method name, then adding W.